scipy/_lib/_util.py

```diff
@@ -14,7 +14,14 @@
 except ImportError:
     class Generator():  # type: ignore[no-redef]
         pass
-
+def asstr(s):
+    if isinstance(s, bytes):
+        return s.decode('latin1')
+    return str(s)
+def asbytes(s):
+    if isinstance(s, bytes):
+        return s
+    return str(s).encode('latin1')
```

Comment:
These are py2-py3 compatibility functions. Given that we only use Python 3, we should not need them. When removing them, the idea is to actually figure out if something is a string or bytes. 

I'll give you one example for `idl.py`.